### PR TITLE
Facilitate local deployment

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
The original deployment configuration requires some parameters and configuration files update when deploying the telematic tool in CAVe-in-a-box. It is complex for a new user to configure it. To facilitate the local deployment and to make it easier to launch this tool in local, we need to automate the local deployment and eliminate as much manually configuration as possible.

Find an issue with web app session during local deployment. After launching the web UI and login in to the UI in a browser, it immediately shows session timeout after successful login. This PR includes the fix to replace the express-session with JWT authentication.

Use predefined influxDB token rather than relying on influxDB randomly generating one.

## Related Issue
https://github.com/usdot-fhwa-stol/cda-telematics/issues/167

## How Has This Been Tested?
Local integration testing

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CDA Telematics Contributing Guide](https://github.com/usdot-fhwa-stol/cda-telematics/blob/main/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: telematic_system/telematic_cloud_messaging/src/main/resources/application.local.properties [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+IGNORE_FIELDS=payload.MessageFrame.value.PersonalSafetyMessage.id
+# Connection timeout to influx bucket. Unit: milliseconds
+INFLUX_CONNECT_TIMEOUT=1000
+# Timeout while writing data to influx. Unit: milliseconds
+INFLUX_WRITE_TIMEOUT=1000
+# MySQL connection properties
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+## Hibernate not make changes to the database
+spring.jpa.hibernate.ddl-auto=none 
+#Replace this URL with actual MYSQL DB host name and database name
```
Comment (adev4a) [inline telematic_system/telematic_cloud_messaging/src/main/resources/application.local.properties:43]: Is this replace still needed for the local configuration?

File: telematic_system/telematic.local.env [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=my-super-secret-auth-token
+INFLUXDB_DEV_ORG=my-org #Required: Create an organization on container startup. You can create more orgs inside the same influxDB container.
+INFLUXDB_DEV_RETENTION=7d #Optional: If not set, default is 0s meaning the initial bucket will retain data forever.
+# NATS server
+NATS_IP=127.0.0.1 #change to amazon ec2 ipv4 address
+NATS_PORT=4222
+##MYSQL 
+MYSQL_USER=telematic
+MYSQL_DATABASE=wfd_grafana
+##Cloud Bridge params
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply telematic_system/telematic.local.env]: Addressed https://github.com/usdot-fhwa-stol/cda-telematics/pull/168/commits/cb1bacccef34b6bdc77bb82bec34e74b5ce1c836

File: telematic_system/telematic.local.env [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# INFLUX DB
+DOCKER_INFLUXDB_INIT_CLI_CONFIG_NAME=telematic
+INFLUXDB_DEV_TAG=latest #Required: The tag for influxDB image version. Current latest version is 2.4 
+INFLUXDB_DEV_INIT_USERNAME=admin #Required: Create a credential username on container initial startup.
+INFLUXDB_DEV_INIT_PASSWORD=P@ssword1 #Required: Create a credential password on container initial startup.
+INFLUXDB_DEV_INIT_BUCKET=infrastructure-dev #Required: Create an bucket on container initial startup. You can create more buckets inside the influxDB container.
+DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=my-super-secret-auth-token
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply telematic_system/telematic.local.env]: Addressed https://github.com/usdot-fhwa-stol/cda-telematics/pull/168/commits/cb1bacccef34b6bdc77bb82bec34e74b5ce1c836

File: telematic_system/telematic.env [hunk 4]
```diff
@@ -1,7 +1,10 @@
+# INFLUX DB
+DOCKER_INFLUXDB_INIT_CLI_CONFIG_NAME=telematic
 INFLUXDB_DEV_TAG=latest #Required: The tag for influxDB image version. Current latest version is 2.4
 INFLUXDB_DEV_INIT_USERNAME=<USERNAME> #Required: Create a credential username on container initial startup.
 INFLUXDB_DEV_INIT_PASSWORD=<PASSWORD> #Required: Create a credential password on container initial startup.
 INFLUXDB_DEV_INIT_BUCKET=platform-dev #Required: Create an bucket on container initial startup. You can create more buckets inside the influxDB container.
+DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=my-super-secret-auth-token
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply telematic_system/telematic.env]: Addressed https://github.com/usdot-fhwa-stol/cda-telematics/pull/168/commits/cb1bacccef34b6bdc77bb82bec34e74b5ce1c836

File: telematic_system/local.setup.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+domain_name_arr=("127.0.0.1 grafana.local.cav-telematics.com"  "127.0.0.1 local.cav-telematics.com"  "127.0.0.1 topic-service.local.cav-telematics.com" "127.0.0.1 ui-service.local.cav-telematics.com")
+
```
[the remark this answers is not in the document]
Comment (dan-du-car) [reply telematic_system/local.setup.sh:8]: Addressed https://github.com/usdot-fhwa-stol/cda-telematics/pull/168/commits/cb1bacccef34b6bdc77bb82bec34e74b5ce1c836

**Comment by paulbourelly999** [comment]
@dan-du-car @SaikrishnaBairamoni I stopped builds on CircleCI for this project (there is no configuration anyways) and I also disabled GitHub status updates for the CircleCI project. We probably want to remove the CircleCI project for this repository all together but this is what I found to do that (https://support.circleci.com/hc/en-us/articles/21040161057947-Can-I-delete-a-project-#:~:text=At%20this%20time%2C%20project%20deletion,remove%20the%20complete%20build%20history.)

Deleting the project can probably be done later. Lets see if the CircleCI configurations I set remove the CircleCI check from this list of checks

**Review by adev4a — APPROVED** [review]

**Review by adev4a — APPROVED** [review]

**Pull request merged**